src/coreComponents/mesh/generators/VTKMeshGenerator.cpp

```diff
@@ -718,6 +1235,8 @@ void fillCellBlock( vtkDataSet & mesh,
   for( vtkIdType c: cellIds )
   {
     vtkCell * const currentCell = mesh.GetCell( c );
+    std::vector< int > const nodeOrder = getGeosxToVtkNodeOrdering( elemType, currentCell );
```

Comment:
We are going to have to split this function into "happy" path (for cell types for which ordering is fixed and can be computed once outside the cell loop) and "sad" path where the order changes from cell to cell. For the latter (or in both cases maybe) we'll want to return the ordering on the stack, using `std::array` or `StackArray` and some upper bound on max number of nodes.

Reply:
Hey folks, I must insist that we address this before the merge, because the change pessimizes common execution path even for meshes that have no `VTK_POLYHEDRON` cells in them, and IMO we should not introduce performance regressions into existing workflows.

Without doing much refactoring, here's a simple solution I propose:
1. Split `getGeosxToVtkNodeOrdering` into two functions, one for fixed-shape VTK cell types (only taking `elemType`), and a separate one for `VTK_POLYHEDRON` (also taking `cell` as input). Basically just put two branches of existing `if/else` into separate functions.
2. Write the cell loop as follows:
```c++
std::vector< int > const nodeOrderFixed = getNodeOrderingFixed( elemType );

localIndex cellCount = 0;
auto const writeCell = [&]( vtkIdType const c, vtkCell * const cell, auto const & nodeOrder )
{
  for( localIndex v = 0; v < numNodesPerElement; v++ )
  {
    cellToVertex[cellCount][v] = cell->GetPointId( nodeOrder[v] );
  }
  localToGlobal[cellCount++] = globalCellId->GetValue( c );
};

for( vtkIdType c: cellIds )
{
  vtkCell * const cell = mesh.GetCell( c );
  if( cell->GetCellType() == VTK_POLYHEDRON )
  {
    writeCell( c, cell, getNodeOrderingPoly( elemType, cell ) );
  }
  else
  {
    writeCell( c, cell, nodeOrderFixed );
  }
}
```

In this case, the dynamic allocation in the loop only occurs for `VTK_POLYHEDRON` cells.

Reply:
The code was modified accordingly.